### ISSUE TITLE
Block access_token creation

### ISSUE THREAD

**Issue by MaxDeg** [IdentityServer/IdentityServer3#2079]
Hi guys,

First of all you did a great job with this project. It's really pleasant to use.

TL;DR; is there a way to return access_denied instead of a token without access to any scope requested?

I'm working on a SPA using identity server with Implicit flow.
But I'm a little bit stuck with the Scope consent. As stated in the spec: http://openid.net/specs/openid-connect-core-1_0.html#Consent
> Once the End-User is authenticated, the Authorization Server MUST obtain an authorization decision before releasing information to the Relying Party. When permitted by the request parameters used, this MAY be done through an interactive dialogue with the End-User that makes it clear what is being consented to or by establishing consent via conditions for processing the request or other means (for example, via previous administrative consent).

The Consent must be obtained for the request. But in identity server only direct user consent is possible. Is there any way to programmatically (based on administrative rules) reject a request? And returning an "access_denied" the same way its done if user don't consent.
For example:
api1 is restrict for particular users only. If an unauthorized user request an access_token for this resource I'm assuming no acces_token must be generated. But if user request access_token for api1 and api2 and can access api2 an access_token must be generated with only acces to api2.
I'm right or not? If i'm not wrong how can I implement that kind of feature?

thanks,

**Comment by leastprivilege** [comment]
There's the ICustomTokenRequestValidator - see if that does what you need.

**Comment by leastprivilege** [comment]
any progress?

**Comment by MaxDeg** [comment]
Yes ICustomTokenRequestValidator able me to customize the accepted scope. 
Thanks a lot ;)
